libfreerdp/core/transport.c

```diff
@@ -1809,3 +1870,191 @@ void transport_set_early_user_auth_mode(rdpTransport* transport, BOOL EUAMode)
 	transport->earlyUserAuth = EUAMode;
 	WLog_Print(transport->log, WLOG_DEBUG, "Early User Auth Mode: %s", EUAMode ? "on" : "off");
 }
+
+rdpTransportLayer* transport_layer_new(rdpTransport* transport, size_t contextSize)
+{
+	rdpTransportLayer* layer =
+	    (rdpTransportLayer*)calloc(1, sizeof(rdpTransportLayer) + contextSize);
+	if (!layer)
+		return NULL;
+
+	if (contextSize)
+		layer->userContext = layer + 1;
```

Comment:
maybe declare a struct at the beginning and allocate that or just allocate the usercontext separately. (makes memory checks easier if something is done out of bound)

Reply:
I'm perfectly ok with @llyzs 's design, just 1 alloc means easier new/free procedures.

Reply:
I made the change, the memory check stuff actually makes sense.